test/Base_Tests/src/Runtime/Stack_Size_Spec.enso

```diff
@@ -0,0 +1,133 @@
+               IO.println 'java_stack_frames='+cnt.to_text
+   ```
+
+   Arguments:
+   - nesting_level How many times should the vector be nested
+generate_code nesting_level:Integer -> Text =
+    bldr = Vector.Builder.new
+    bldr.append "from Standard.Base import all"
+    bldr.append "polyglot java import java.lang.RuntimeException"
+    bldr.append '\n'
+    bldr.append <| "main = "
+    bldr.append <| "    "
+                   + "vec = "
+                   + ("[" * nesting_level)
+                   + "42"
+                   + ("]" * nesting_level)
+    bldr.append <| "    "
+                   + "vec.map e0->"
+    0.up_to (nesting_level - 1) . each \i ->
+        bldr.append <| ("    " * (i + 2))
+                       + "e"
+                       + i.to_text
+                       + ".map e"
+                       + (i + 1).to_text
+                       + "-> "
+    bldr.append <| ("    " * (nesting_level + 1))
+                   + "cnt = RuntimeException.new.getStackTrace.length"
+    bldr.append <| ("    " * (nesting_level + 1))
+                   + "IO.println <| 'java_stack_frames=' + cnt.to_text"
+                   + '\n'
+    bldr.to_vector.reduce \first_line:Text second_line:Text ->
+        first_line + '\n' + second_line
+
+
+## Runs Enso subprocess with disabled Truffle compiler, with
+   larger thread stack and also with larger stack trace element collected
+   (which is needed for `new RuntimeException().getStackTrace().length`)
+   as this value is by default set only to 1024.
+
+   The thread stack size is also set to a sufficiently large value
+   to ensure there is no StackOverflow.
+run_without_compiler enso_args:Vector -> Process_Result =
+    java_opts = "-Dpolyglot.engine.Compilation=false "
+                + "-XX:MaxJavaStackTraceDepth=18000 "
+                + "-Xms16M"
+    args = ["JAVA_OPTS="+java_opts, enso_bin.path] + enso_args
+    Process.run "env" (args + enso_args)
+
+
+## Runs enso as a subprocess with the specified nesting level of `Vector.map` calls.
+   Returns count of Java stack frames from the deepest `Vector.map` call.
+
+   Arguments:
+   - nesting Level of nesting of `Vector.map` method calls.
+run nesting:Integer -> Integer =
+    tmp_file = File.create_temporary_file suffix=".enso"
+    code = generate_code nesting
+    code.write tmp_file
+    proc_res = run_without_compiler ["--run", tmp_file.path]
+    # FInd and parse a specific line from the process stdout
+    j_frames_line = proc_res.stdout.split '\n' . find \line ->
+        line.contains "java_stack_frames"
+    j_frames_line.split '=' . last . parse_integer
+
+
+add_specs suite_builder =
+    suite_builder.group "Stack size" \group_builder ->
+        group_builder.specify "Java stack size of nested Vector.map should be kept reasonably low" <|
+            nesting_10 = run 10
+            nesting_11 = run 11
+            stack_size = nesting_11 - nesting_10
+            (stack_size < 115) . should_be_true
```

Comment:
According to [following note](https://github.com/enso-org/enso/pull/11363#issuecomment-2432996902):

> the number of stack frames is 22 (or) ... 35 Java stack frames per `Vector.map` method call is sufficient improvement.

Shouldn't this number be at most `40`?

Reply:
In e0063aab6596fcdcdb7298e2d05250c63e243866, the `Vector.map` and `Array_Like_Helpers.map` are marked as `@Tail_Call`. Moreover, the test specifies `40` and succeeds.